consumer-java/src/main/java/org/apache/pulsar/client/impl/WeightedMultiTopicsConsumerImpl.java

```diff
@@ -114,7 +115,7 @@
                                     ConsumerInterceptors<T> interceptors, boolean createTopicIfDoesNotExist, MessageId startMessageId,
                                     long startMessageRollbackDurationInSec) {
         super(client, singleTopic, conf, Math.max(2, conf.getReceiverQueueSize()), executorProvider, subscribeFuture,
-                schema, interceptors);
+                schema, interceptors);// Why at max 2 is hardcoded here
```

Comment:
It's not max 2, it is a max of two and the given receiver queue size, based on this the consume permits of the consumers are determined

Reply:
got it.. because the permits are half the receiver queue size.